File: Unidad 2/Lista Doble en C++/ListaDoble.cpp
#include <iostream.h>
#include <conio.h>

class Nodo{
   private:
   int dato;
   Nodo *sig.*ant;

public:
   Nodo(){dato=0; sig=ant=NULL;}

   Nodo(int d){dato=; sin=ant=NULL;}

   ~Nodo(){cout<<"Memoria Liberada";}

   int getDato(){
      return dato;}

   void setDato(int d){dato=d;}

   Nodo * getSig(){
      return sig;}

   void setSig(Nodo *s){sig=s;}

   Nodo * getAnt(){
      return ant;}

   void setSig(Nodo *a){ant=a;}

};



class ListaDoble{
  private:
   Nodo *inicio,*fin;
  public:
  
   ListaDoble(){inicio=fin=NULL;}
   
   Nodo* getMayorque(int d){
      for(Nodo *i=inicio->getSig(); i!=NULL; i=i->getSit()){
         if(i->getDato()>d)
            return i;
         return NULL;
      
      
      }
   
   }//Mayor que para saber que show para ordenar
   
   
   
   
   
   void InsertarOrd(int d){
      if(inicio==NULL){inicio=fin=new Nodo(d);}
      else if(d<=inicio->getDato()){
         inicio->setAnt(new Nodo(d));
         inicio->getAnt().setSig(inicio);
         inicio=inicio->getAnt();
      }
      else{
         Nodo * temp=getMayorque(d);
         if(temp==NULL){
            fin->setSig(new Nodo(d));
            fin->getSig()->setAnt(fin);
            fin=fin->getSig();
         else{
          Nodo *aux=new Nodo(d);
          aux->setSig(temp);
          aux->setAnt(temp->getAnt());
          temp->getAnt()->setSig(aux);
          aux->getAnt()->setSig(aux);
          
         
         }
     
         
         }
      
      }//else
   }//Insertar
   
   void verLista(){
   
   
   
   }//ver lista
   
   
   
   ~ListaDoble(){
   
   
   }//Destructr

};

void main(){
   ListaDoble list;
   list.InsertarOrd(10);
   list.InsertarOrd(3);
   list.verLista;
   list.InsertarOrd(7);
   list.InsertarOrd(11);
}

File: Unidad 2/ListaDoble/Racional/ListaDoble.java
import static javax.swing.JOptionPane.*;
class ListaDoble{   //Inicio de la clase
   private NodoD inicio, fin;

   public ListaDoble(){inicio=null;fin=null;}

   public void agregar(Racional d){
      if(inicio==null)
         fin=inicio=new NodoD(d);
      else{
         NodoD temp=new NodoD(d);
         fin.setSig(temp);
         temp.setAnt(fin);
         fin=fin.getSig();
      }//else
   }//Agregar  
   
   public NodoD buscar(Racional d){
      for(NodoD i=inicio; i!=null; i=i.getSig())
         if(i.getDato().igualigual(d))
            return i;
      return null;
   }//buscar
   
   
   public void eliminar(Racional d){
      if(inicio.getDato().igualigual(d)){inicio=inicio.getSig(); 
         inicio.setAnt(null);
         return;}
         
      if(fin.getDato().igualigual(d)){
         fin=fin.getAnt();
         fin.setSig(null);
         return;
      }
      
      NodoD temp=buscar(d);
      if(temp!=null){
         temp.getAnt().setSig(temp.getSig());
         temp.getSig().setAnt(temp.getAnt());
      }
      else showMessageDialog(null,"El dato que quiere eliminar no existe");
   }//eliminar
    
   public Racional Suma(){
      NodoD R=inicio;
      Racional aux=new Racional();
      while(R!=null){
         aux=aux.mas(R.getDato());
         R=R.getSig();
      }
   
      return aux;
   }//sumar
  
   public Racional mayor(){
      Racional aux=new Racional();
      for(NodoD i=inicio; i!=null; i=i.getSig())
         if(i.getDato().mayor(aux))aux=i.getDato();
      return aux;
   }//mayor
    
   
   public Racional menor(){
      Racional aux=mayor();
      for(NodoD i=inicio; i!=null; i=i.getSig())
         if(i.getDato().menor(aux))aux=i.getDato();
      return aux;
   }//menor
   
   public String pares(){
      String aux="";
      for(NodoD i=inicio; i!=null; i=i.getSig())
         if(i.getDato().div(2).igualigual(0)){aux=aux+i.getDato()+" , ";}
      return aux;
   }//pares
   
   public String inpares(){
      String aux="";
      for(NodoD i=inicio; i!=null; i=i.getSig())
         if(i.getDato().div(2).diferente(0)){aux=aux+i.getDato()+" , ";}
      return aux;
   }//inpares
   

   public void verLista(){
      String cad="";
      for(NodoD i=inicio; i!=null; i=i.getSig())
         cad=" "+cad+"[ "+i+" ]";
      showMessageDialog(null,cad+"\n Suma: "+Suma()+"\n Menor: "+menor()+"\n Mayor: "+mayor()+
         "\n Pares: "+pares()+"\n inPares: "+inpares());//*/
      //showMessageDialog(null, cad);
   }//ver lista




   public static void main(String trs[]){
      ListaDoble list=new ListaDoble();
      list.agregar(new Racional(2,4)); list.agregar(new Racional(3,4)); list.agregar(new Racional(1,2));
      list.verLista();
      list.eliminar(new Racional(1,2));
      list.verLista();
   }//main
}//Class